### DISCUSSION TITLE
Encrypted data communication

### DISCUSSION THREAD

**Discussion by tbobek** [jomjol/AI-on-the-edge-device#1502] [Ideas]
Hi! 

first of all: this is a great project and I really love the approach and the functionality! 

My question would be: is it possible to make data communication encrypted? I've actually modified the project to use https and 
also mqtts. The build and deployment process is finished without errors, but during a page request, an "out of memory" error is 
emitted on a standard ESP32-CAM device. To be able to solve the problem, I would see three approaches: 

1. use another device with more memory. I've googled, but didn't found an appropriate device. Has anyone a hint? 
2. reduce the code to make it work on the ESP32-CAM device. Which parts of the code could be omitted? 
3. another person, much smarter than me, has already developed a solution for that. Can someone point me to that solution? 

Thanks in advance for any input on that!

**Comment by jomjol** [comment]
This is a frequent question in the forum. We decided not to go for encryption, because you would need a real safety and security concept.

Anything, that would use here would only lead somebody to believe in the security and this cannot be maintained within the freetime of the developers.
